pkg/proto/transactions_with_proofs.go

```diff
@@ -187,8 +187,9 @@ func NewUnsignedIssueWithProofs(v byte, senderPK crypto.PublicKey, name, descrip
 	return &IssueWithProofs{Type: IssueTransaction, Version: v, Script: script, Issue: i}
 }
 
-func (tx *IssueWithProofs) Validate(_ Scheme) (Transaction, error) {
-	if tx.Version < 2 || tx.Version > MaxIssueTransactionVersion {
+func (tx *IssueWithProofs) Validate(params TransactionValidationParams) (Transaction, error) {
+	if tx.Version < 2 || params.CheckVersion && tx.Version > MaxIssueTransactionVersion ||
+		!params.CheckVersion && tx.Version > MaxUncheckedTransactionVersion {
 		return tx, errors.Errorf("unexpected version %d for IssueWithProofs", tx.Version)
 	}
```

Comment:
These checks are similar. The common function can be used for tx version check. Take a look:
```go
func checkTxVersion(tx Transaction, expectedV byte, checkVersion bool) error {
	v := tx.GetVersion()
	if v < 1 || checkVersion && v > expectedV || !checkVersion && v > MaxUncheckedTransactionVersion {
		return errors.Errorf("unexpected version for (%T): want %d, got %d", tx, expectedV, v)
	}
	return nil
}
```

Reply:
It's code generation made by hand. No need to add function.

Reply:
What do you mean "by hand"? The code in this file is not generated. Hence, the whole code in this file can't be considered as generated. Generated code (imho) should be placed into a separate file.

Reply:
Don't know what to say, I mean that this code resembles generated. Of course, I didn't tell that it generated.